Добавление анимаций в проект

zklevsha opened the pull request:
---
:mortar_board: [Добавление анимаций в проект](https://up.htmlacademy.ru/vue/1/user/1964457/tasks/7)

Commit:
Added PopUp animation

Commit:
Added animation to pages transition

Commit:
Added ingredient animation

Commit:
bugfix with empty orders after page refresh (not related to this task)

Djaler commented on src/frontend/src/App.vue:
```diff
@@ -1,7 +1,13 @@
+      <transition
+        name="view"
+        appear
+        enter-active-class="animate__animated animate__slideInRight"
```
эта анимация не совсем корректно работает с окном входа. это происходит из-за того, что в этой анимации используется `transform: translate3d`, а в стилях окна входа - `transform: translate`. Он там используется для выравнивания по центру экрана. Предлагаю починить это, поправив стили окна входа (`.sign-form`), сделав выравнивание по центру другим способом, например с помощью `margin: auto`

Djaler commented on src/frontend/src/views/Cart.vue:
```diff
@@ -114,7 +114,7 @@ export default {
       this.$store.dispatch("Cart/resetState");
       this.$store.commit("Builder/resetState");
-      this.$router.push("/");
+      setTimeout(() => this.$router.push("/"), 200); // waiting on PopUp faidout animation
```
чтобы не хардкодить тут предполагаемую длительность анимации, можно воспользоваться специальными хуками, позволяющими отследить реальное завершение анимации https://ru.vuejs.org/v2/guide/transitions.html#JavaScript-%D1%85%D1%83%D0%BA%D0%B8

Commit:
1. Fixed animation for login window
2. CartPopUp: get rid of setTimeout if favor of transition hooks
3. Added close action to Login window (not related to this task)
